src/main/java/jaci/openrio/toast/lib/registry/Registrar.java

```diff
@@ -32,6 +32,15 @@
      */
     public synchronized  <T extends Type> T fetch(ID id, Class<T> clazz, Supplier<T> creator) {
         Type in_register = registered.get(id);
+
+        if(in_register == null) {
+            for (ID key : registered.keySet()) {
+                if (key.equals(id)) {
+                    in_register = registered.get(key);
+                }
+            }
+        }
+
         if (in_register == null) {
```

Comment:
They have different functions. I suppose it _could_ be combined, but the whole fetch method would have to be restructured. Think of this commit's code as an expanded version of line 34. If the hash comparison failed, we run a comparison against values. If that fails, we move on and create a new instance of the object.